src/FlightDisplay/GuidedActionsController.qml

```diff
@@ -105,9 +105,12 @@ Item {
     readonly property int actionStepAlt:                    25
 
     property var    _activeVehicle:             QGroundControl.multiVehicleManager.activeVehicle
+    property var    _planMasterController:      globals.planMasterControllerPlanView
+    property bool   _syncInProgress:            _planMasterController.syncInProgress
+    property bool   _dirty:                     _planMasterController.dirty
     property bool   _useChecklist:              QGroundControl.settingsManager.appSettings.useChecklist.rawValue && QGroundControl.corePlugin.options.preFlightChecklistUrl.toString().length
     property bool   _enforceChecklist:          _useChecklist && QGroundControl.settingsManager.appSettings.enforceChecklist.rawValue
-    property bool   _canArm:                    _activeVehicle ? (_useChecklist ? (_enforceChecklist ? _activeVehicle.checkListState === Vehicle.CheckListPassed : true) : true) : false
+    property bool   _canArm:                    canArm()
```

Comment:
This does not actually block arming, but hides the "guided mission" button. It might be OK, but I think it is perhaps better to show the drone as "not ready" (yellow).
Also, when we add things that blocks arming/flight ready/take-off, it is very important that we visualise for the pilot why it is blocked.

Reply:
That can work, but we probably need more "OPS friendly" names.
"Syncing" and "dirty" makes sense in the code, but "Upload failed" and similar is probably better in GUI.

Reply:
True, will fix

Reply:
Changed "dirty" to "Mission not uploaded to vehicle". The problem with replacing "Syncing" is that the `syncInProgress` variable represents when transactions between drone and qgc is happening, which could be both uploading and downloading missions, I think it is hard to distinguish upload and download in the current codebase. I think "syncing" is descriptive enough, especially as it is accompanied with the progress bar.
[Screencast from 23. okt. 2024 kl. 19.58 +0200.webm](https://github.com/user-attachments/assets/bf7dd798-2d42-4183-91d1-68ba85d61eea)